superx/information_extractors/item_info_extractor.py

```diff
@@ -148,24 +130,19 @@ def extract_information_from_parsed_xml(self, xml_info_child_node):
             if item.find(is_weighted_attr).text == '1':
                 is_weighted = True
 
-            unit_of_measure = self.standardize_weight_name(item.find('UnitQty').text)
-            # if item already in db then continue to next item
-            if bool(Product.query.filter_by(id=item_code).first()):
-                # adding current_branch_price to the db and commit
-                current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
-                db.session.add(current_branch_price)
-                db.session.commit()
-                continue
-
-            # adding current_product to the db and commit
-            current_product = Product(id=item_code, name=item_name, quantity=quantity, is_weighted=is_weighted,
-                                      unit_of_measure=unit_of_measure)
-            db.session.add(current_product)
-            db.session.commit()
-            # adding current_branch_price to the db and commit
-            current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
-            db.session.add(current_branch_price)
-            db.session.commit()
+            unit_of_measure = self.standardize_weight_name(item.find('UnitQty').text.strip())
+            # if item is not in db then add it
+            if not bool(Product.query.filter_by(id=item_code).first()):
+                current_product = Product(id=item_code, name=item_name, quantity=quantity, is_weighted=is_weighted,
+                                          unit_of_measure=unit_of_measure)
+                db.session.add(current_product)
+
+            branch_price_list.append(BranchPrice(branch_id=branch_id, item_code=item_code, price=price,
+                                                 update_date=update_date))
+
+        db.session.commit()
+        db.session.add_all(branch_price_list)
+        db.session.commit()
```

Comment:
2 separate commits are necessary as branch_price has a foreign key from the products table, and so firt all the products must be committed before the branch _prices are committed

Reply:
i will try it put, the list object is very costly